### ISSUE TITLE
Cannot build a target for linux arm64

### ISSUE THREAD

**Issue by seungduk-yanolja** [uber/bazel-zig-cc#13]
Hello there,

First of all, thanks for the great work.
Using the following rule, I am trying to integrate a CPP library into a Go package.

```
# This rule is defined for each architecture in each directory.
cc_import(
    name = "banana_lib",
    static_library = "libbanana_static.a",
)

go_library(
    name = "banana",
    srcs = glob(["include/**/*.h"]) + [
        "cfunc.go",
        "banana.go",
    ],  # keep
    cdeps = select({
        "@io_bazel_rules_go//go/platform:linux_arm64": [
            "//test/banana/linux_arm64:banana_lib",
        ],
        "@io_bazel_rules_go//go/platform:linux_amd64": [
            "//test/banana/linux_amd64:banana_lib",
        ],
        "@io_bazel_rules_go//go/platform:darwin_amd64": [
            "//test/banana/darwin_x86_64:banana_lib",
        ],
        "@io_bazel_rules_go//go/platform:darwin_arm64": [
            "//test/banana/darwin_arm64:banana_lib",
        ],
        "//conditions:default": [],
    }),
    cgo = True,
    clinkopts = ["-lpthread -lstdc++"],
    data = [":banana_dict"],
    importpath = "seungduk.kim/test/banana",
)
```

It requires a toolchain, so I imported `bazel-zig-cc` in WORKSPACE.

```
BAZEL_ZIG_CC_VERSION = "v1.0.1"

http_archive(
    name = "bazel-zig-cc",
    sha256 = "e9f82bfb74b3df5ca0e67f4d4989e7f1f7ce3386c295fd7fda881ab91f83e509",
    strip_prefix = "bazel-zig-cc-{}".format(BAZEL_ZIG_CC_VERSION),
    urls = ["https://git.sr.ht/~motiejus/bazel-zig-cc/archive/{}.tar.gz".format(BAZEL_ZIG_CC_VERSION)],
)

load("@bazel-zig-cc//toolchain:defs.bzl", zig_toolchains = "toolchains")

zig_toolchains()
```

And I ran the following command.

```
 bazel build --platforms @zig_sdk//platform:linux_arm64 --extra_toolchains @zig_sdk//toolchain:linux_arm64_gnu.2.28  --incompatible_enable_cc_toolchain_resolution  //test/banana
```

Running the command above resulted the following.

```
ld.lld: error: undefined symbol: std::string::compare(char const*) const
>>> referenced by ArchUtils.cpp
>>>               ArchUtils.cpp.o:(parseArchType(char const*)) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a

ld.lld: error: undefined symbol: std::_Rb_tree_increment(std::_Rb_tree_node_base const*)
>>> referenced by ModelDetector.cpp
>>>               ModelDetector.cpp.o:(banana::ModelDetector::saveModel(std::string const&) const) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a
>>> referenced by ModelDetector.cpp
>>>               ModelDetector.cpp.o:(banana::ModelDetector::saveModel(std::string const&) const) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a
>>> referenced by ModelDetector.cpp
>>>               ModelDetector.cpp.o:(banana::ModelDetector::saveModel(std::string const&) const) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a
>>> referenced 8 more times

ld.lld: error: undefined symbol: std::_Rb_tree_rebalance_for_erase(std::_Rb_tree_node_base*, std::_Rb_tree_node_base&)
>>> referenced by ModelDetector.cpp
>>>               ModelDetector.cpp.o:(banana::ModelDetector::countNgram(banana::ModelDetector::Counter&, std::function<std::basic_string<char16_t, std::char_traits<char16_t>, std::allocator<char16_t>> ()> const&, unsigned long, unsigned long) const) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a
>>> referenced by ModelDetector.cpp
>>>               ModelDetector.cpp.o:(banana::ModelDetector::countNgram(banana::ModelDetector::Counter&, std::function<std::basic_string<char16_t, std::char_traits<char16_t>, std::allocator<char16_t>> ()> const&, unsigned long, unsigned long) const) in archive experimental/seungduk.kim/test/banana/linux_arm64/libbanana_static.a
compilepkg: error running subcommand external/zig_sdk/tools/aarch64-linux-gnu.2.28/c++: exit status 1
Target //experimental/seungduk.kim/test:test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 32.203s, Critical Path: 31.75s
INFO: 5 processes: 2 internal, 3 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

I also tried @zig_sdk//toolchain:linux_arm64_musl but nothing worked. I am using a darwin_arm64 machine and every targets of darwin (darwin_arm64, darwin_amd64) works well.
The error messages seem like all standard libraries failed to be linked. Any thoughts?

Thank you!

**Comment by motiejus** [comment]
Here is the crux of the problem: https://github.com/ziglang/zig/issues/9832#issuecomment-926832810

TLDR: compile your dependencies with `zig c++` too.

**Comment by seungduk-yanolja** [comment]
No easy job there... Thanks a lot!
